### PR TITLE
[InternLM] Add support for InternLM

### PR DESCRIPTION
**Pull request by Rocketknight1**
InternLM is based on the LLaMA code but adds a `config.bias` parameter. We can support those models by adding `config.bias` to LLaMA, and preserve backward compatibility by defaulting it to `False`

### CODE REVIEW

File: src/transformers/models/llama/configuration_llama.py [hunk 1]
```diff
@@ -125,6 +125,7 @@ def __init__(
         tie_word_embeddings=False,
         rope_theta=10000.0,
         rope_scaling=None,
+        bias=False,
```
Comment (ArthurZucker) [inline src/transformers/models/llama/configuration_llama.py:128]: let's document this, and use `attention_bias` or something more expressive

**Review by ArthurZucker — COMMENTED** [review]
Thanks! Quick question, do they use `dropout` after attention? 
Let's also add a `InternLM.md`, add it to the doc etc 😉

File: src/transformers/models/llama/configuration_llama.py (hunk 1 above)
Comment (Rocketknight1) [reply src/transformers/models/llama/configuration_llama.py:128]: Sure!

Comment (ArthurZucker) [reply src/transformers/models/llama/configuration_llama.py:128]: On the same note the PR could be rename to '[`InternLM`] Add support for InternLM' 😉

Comment (Rocketknight1) [reply src/transformers/models/llama/configuration_llama.py]: Both done!

**Comment by Rocketknight1** [comment]
@ArthurZucker Also, to answer the question, there is no `Dropout` anywhere in the LLaMA code or the InternLM code.

**Review by LysandreJik — APPROVED** [review]

**Pull request merged**